executor/runtime/docker/docker.go

```diff
@@ -1503,23 +1506,21 @@ func getMainContainerRoot(inspectOutput types.ContainerJSON) string {
 	return inspectOutput.GraphDriver.Data["MergedDir"]
 }
 
-func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, c runtimeTypes.Container, eventChan <-chan events.Message, errChan <-chan error, statusMessageChan chan runtimeTypes.StatusMessage) {
-	defer close(statusMessageChan)
-	defer cancel()
-
-	// This context should be tied to the lifetime of the container -- it will get significantly less broken
-	// when we tear out the launchguard code
-
+func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, containerID string, eventChan <-chan events.Message, errChan <-chan error, statusMessageChan chan runtimeTypes.StatusMessage) {
 	for {
 		// 3. If the current state of the container is terminal, send it, and bail
 		// 4. Else, keep sending messages until we bail
 		select {
 		case err := <-errChan:
-			log.Fatal("Got error while listening for docker events, bailing: ", err)
+			if errors.Is(err, context.Canceled) {
+				return
+			}
+			log.Fatalf("Got unexpected error while listening for docker events for %s, bailing: %s", containerID, err)
```

Comment:
Maybe we should log error here and return. Why kill the executor with fatal?

Reply:
I agree.